lib/timers.js

```diff
@@ -316,6 +319,11 @@ exports.setTimeout = function(callback, after) {
 
   after *= 1; // coalesce to number or NaN
 
+  if (after > TIMEOUT_MAX) {
```

Comment:
> I'm going off the fact that the max timeout might be surprising.

The warning will anyway show `NaN does not fit...`. That would already be surprising I guess